src/test.cairo

```diff
@@ -0,0 +1,576 @@
+use openzeppelin::account::AccountComponent::{InternalTrait, SRC6CamelOnlyImpl};
+use openzeppelin::account::AccountComponent::{OwnerAdded, OwnerRemoved};
+use openzeppelin::account::AccountComponent::{PublicKeyCamelImpl, PublicKeyImpl};
+use openzeppelin::account::AccountComponent;
+use openzeppelin::account::interface::{AccountABIDispatcherTrait, AccountABIDispatcher};
+use openzeppelin::account::interface::{ISRC6, ISRC6_ID};
+use openzeppelin::introspection::interface::{ISRC5, ISRC5_ID};
+
+
+use agentstark::account_mock::DualCaseAccountMock;
+use agentstark::erc20_mock::DualCaseERC20Mock;
+//use openzeppelin::tests::mocks::account_mocks::DualCaseAccountMock;
+//use openzeppelin::tests::mocks::erc20_mocks::DualCaseERC20Mock;
+use openzeppelin::tests::utils::constants::{
+    NAME, SYMBOL, PUBKEY, NEW_PUBKEY, SALT, ZERO, QUERY_OFFSET, QUERY_VERSION,
+    MIN_TRANSACTION_VERSION
+};
+use openzeppelin::tests::utils;
+use openzeppelin::token::erc20::interface::{IERC20DispatcherTrait, IERC20Dispatcher};
+use openzeppelin::utils::selectors;
+use openzeppelin::utils::serde::SerializedAppend;
+use starknet::ContractAddress;
+use starknet::account::Call;
+use starknet::contract_address_const;
+use starknet::testing;
+
+#[derive(Drop)]
+struct SignedTransactionData {
+    private_key: felt252,
+    public_key: felt252,
+    transaction_hash: felt252,
+    r: felt252,
+    s: felt252
+}
+
+fn SIGNED_TX_DATA() -> SignedTransactionData {
+    SignedTransactionData {
+        private_key: 1234,
+        public_key: 0x1f3c942d7f492a37608cde0d77b884a5aa9e11d2919225968557370ddb5a5aa,
+        transaction_hash: 0x601d3d2e265c10ff645e1554c435e72ce6721f0ba5fc96f0c650bfc6231191a,
+        r: 0x6c8be1fb0fb5c730fbd7abaecbed9d980376ff2e660dfcd157e158d2b026891,
+        s: 0x76b4669998eb933f44a59eace12b41328ab975ceafddf92602b21eb23e22e35
+    }
+}
+
+//
+// Constants
+//
+
+
+use hash::Hash;
+
+fn CLASS_HASH() -> felt252 {
+    starknet_keccak(
```

Comment:
You need to use `keccak::cairo_keccak(ref data, n.into(), 1)` such as in [here](https://github.com/HerodotusDev/integrity/blob/a29acce9aa3b3f52c8fdabdcb7008dc5db364816/src/common/hasher.cairo#L24)

Right now you are trying to use pure Rust code from Rust crates, hence it doesn't work

Reply:
Or actually, let's just use constant values for the contracts we test and use?